### PR TITLE
Deprecate getResults with arguments

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

**Comment by johnktims** [comment]
This looks like a fine change.

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java [hunk 1]
```diff
@@ -251,10 +251,23 @@ protected void endContext(SQLListenerContext context) {
...
     public ResultSet getResults(Expression<?>... exprs) {
-        queryMixin.setProjection(exprs);
+        if (exprs.length > 0) {
+            queryMixin.setProjection(exprs);
+        }
+        return getResults();
+    }
 
+    /**
+     * Get the results as an JDBC result set
```
Comment (Shredder121) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java]: I would turn this into
`Get the results as a JDBC result set`
or
`Get the results as a JDBC ResultSet`

**Comment by Shredder121** [comment]
I made a small adjustment to the Javadoc, to me this looks good now.
@johnktims can you look one more time and merge if it's okay?

**Comment by johnktims** [comment]
Looks good to me.

**Pull request merged**